helm-chart/enroute/templates/serviceaccount.yaml

```diff
@@ -1,80 +1,12 @@
-{{- if .Values.enrouteService.rbac.create -}}
+{{- if .Values.serviceAccount.create -}}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: enroutedemo
-  namespace: {{ .Values.globalSettings.namespace.name }}
+  name: {{ include "enroute.serviceAccountName" . }}
```

Comment:
Do we need the namespace here -

```
namespace: {{ .Release.Namespace }}
```

Note, that the ```ClusterRoleBinding``` references the ```ServiceAccount``` along with namespace on line 67 in roles.yaml

Reply:
The namespace is set implicitly by helm. Eg the command `helm template -n test someenroute helm-chart/enroute-0.5.0.tgz` contains the NS `test` and everything will be deployed to that namespace. If you have the need to deploy object to not the "default" NS defined in the helm command, then you can set an explicit namespace. But its not required by default.

The ClusterRoleBinding is a clusterwide resource and SA are namespaced. Thats why we need to set the namspace there explicitly.